Tests/tests/test_ActivityMonitor.cpp

```diff
@@ -0,0 +1,219 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+#include "gtest/gtest.h"
+
+#include "ActivityMonitor.h"
+
+#include "FactoriesImplementation.h"
+#include "ServiceMock.h"
+#include "IarmBusMock.h"
+#include <fstream>
+#include <unistd.h>
+
+using namespace WPEFramework;
+
+namespace {
+const string regFile = _T("/opt/waylandregistry.conf");
+const uint8_t regFileData[] = "{\"waylandapps\":[{\"name\" : \"memcheck-amd64-\",\"binary\" : \"/usr/lib/x86_64-linux-gnu/valgrind/memcheck-amd64-%\"}]}";
+}
+
+class ActivityMonitorTest : public ::testing::Test {
+protected:
+    Core::ProxyType<Plugin::ActivityMonitor> plugin;
+    Core::JSONRPC::Handler& handler;
+    Core::JSONRPC::Connection connection;
+    string response;
+
+    ActivityMonitorTest()
+        : plugin(Core::ProxyType<Plugin::ActivityMonitor>::Create())
+        , handler(*(plugin))
+        , connection(1, 0)
+    {
+        EXPECT_EQ(string(""), plugin->Initialize(nullptr));
+    }
+    virtual ~ActivityMonitorTest() override
+    {
+        IarmBus::getInstance().impl = nullptr;
+        plugin->Deinitialize(nullptr);
+    }
+};
+
+TEST_F(ActivityMonitorTest, registeredMethods)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("enableMonitoring")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("disableMonitoring")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getApplicationMemoryUsage")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getAllMemoryUsage")));
+}
+
+TEST_F(ActivityMonitorTest, disableMonitoring)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disableMonitoring"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+}
+
+TEST_F(ActivityMonitorTest, enableMonitoringEmptyConfig)
+{
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("enableMonitoring"), _T("{\"config\":[],\"memoryIntervalSeconds\":\"0\", \"cpuIntervalSeconds\":\"0\"}"), response));
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("enableMonitoring"), _T("{\"config\":[{\"appPid\": 6763, \"memoryThresholdMB\": 10,\"cpuThresholdPercent\": 50,\"cpuThresholdSeconds\": 2}],\"memoryIntervalSeconds\":\"0\", \"cpuIntervalSeconds\":\"0\"}"), response));
+}
+
+class ActivityMonitorMemoryUsageTest : public ActivityMonitorTest
+{
+protected:
+    IarmBusImplMock iarmBusImplMock;
+    ActivityMonitorMemoryUsageTest() : ActivityMonitorTest()
+    {
+        //Add version info
+        std::ofstream fileVer("/version.txt");
+        fileVer << "imagename:PX051AEI_VBN_2203_sprint_20220331225312sdy_NG\nSDK_VERSION=17.3\nMEDIARITE=8.3.53\nYOCTO_VERSION=dunfell\n";
+        fileVer.close();
+
```

Comment:
it's not clear why version.txt is needed for tests. can you please check if it works without this file?

Reply:
The function "isDevOrVBNImage" uses the image name to check if it is PROD or DEV.

Reply:
> can you please check if it works without this file

Reply:
I verified and without version.txt:
- getAllMemoryUsage() - Returns empty application memory data as "r`esponse={"freeMemoryMB":7198,"applicationMemory":[],"success":true}`"
- enableMonitoring() - Returns "`ERROR [ActivityMonitor.cpp:899] monitoring: Failed to determine memory usage for 4808`"